File: benchmarks/cross-language-v1/tasks/owned-byte-sentinel-balance-v1/public/typescript/candidate.ts
export function sentinelChecksum(input: Uint8Array): number {
  const transformed = new Uint8Array(input);
  for (let index = 0; index < transformed.length; index += 1) {
    const byte = transformed[index];
    if (byte === 0xff) {
      transformed[index] = 0x00;
    } else if (byte === 0x00) {
      transformed[index] = 0xff;
    } else {
      transformed[index] = 0x01;
    }
  }
  let checksum = 0;
  for (let index = 0; index < transformed.length; index += 1) {
    checksum += (index + 1) * transformed[index];
  }
  return checksum;
}

File: benchmarks/cross-language-v1/tasks/owned-byte-sentinel-balance-v1/public/rust/candidate.rs
pub fn sentinel_checksum(input: &[u8]) -> usize {
    let mut transformed = input.to_vec();
    for byte in &mut transformed {
        if *byte == 0xff {
            *byte = 0x00;
        } else if *byte == 0x00 {
            *byte = 0xff;
        } else {
            *byte = 0x01;
        }
    }
    transformed
        .iter()
        .enumerate()
        .map(|(index, byte)| (index + 1) * usize::from(*byte))
        .sum()
}
